itest/rust/src/register_tests/derive_variant_test.rs

```diff
@@ -7,261 +7,104 @@
+#[derive(GodotConvert, Clone, PartialEq, Debug)]
+#[godot(via = i64)]
+enum EnumInty {
+    A = 10,
+    B,
+    C,
+    D = 1,
+    E,
 }
 
 #[itest]
-fn new_type_struct() {
-    roundtrip(StructNewType(String::from("five")));
-    roundtrip(dict! { "StructNewType" : "five" })
+fn newtype_tuple_struct() {
+    roundtrip(TupleNewtype("hello!".into()));
 }
 
 #[itest]
-fn tuple_struct() {
-    roundtrip(StructTuple(String::from("one"), 2));
-    roundtrip(dict! {
-        "StructTuple": varray!["one", 2]
+fn newtype_named_struct() {
+    roundtrip(NamedNewtype {
+        field1: Vector2::new(10.0, 25.0),
     });
 }
 
 #[itest]
-fn named_struct() {
-    roundtrip(StructNamed {
-        field1: String::from("four"),
-        field2: 5,
-    });
-    roundtrip(dict! {
-        "StructNamed": dict! { "field1": "four", "field2": 5 }
-    });
+fn enum_stringy() {
+    roundtrip(EnumStringy::A);
+    assert_eq!(EnumStringy::A.to_godot(), "A".into());
+    roundtrip(EnumStringy::B);
+    assert_eq!(EnumStringy::B.to_godot(), "B".into());
+    roundtrip(EnumStringy::C);
+    assert_eq!(EnumStringy::C.to_godot(), "C".into());
+    roundtrip(EnumStringy::D);
+    assert_eq!(EnumStringy::D.to_godot(), "D".into());
 }
 
 #[itest]
-fn generics() {
-    roundtrip(StructGenWhere(String::from("4")));
-    roundtrip(dict! { "StructGenWhere": "4" });
-}
-
-impl Bound for String {}
-
-#[itest]
-fn generics_bound() {
-    roundtrip(StructGenBound(String::from("4")));
-    roundtrip(dict! { "StructGenBound": "4" });
-}
-
-#[itest]
-fn enum_unit() {
-    roundtrip(Enum::Unit);
-    roundtrip(dict! { "Enum": "Unit" });
-}
-
-#[itest]
-fn enum_one_tuple() {
-    roundtrip(Enum::OneTuple(4));
-    roundtrip(dict! {
-        "Enum": dict! { "OneTuple" : 4 }
-    });
-}
-
-#[itest]
-fn enum_tuple() {
-    roundtrip(Enum::Tuple(String::from("four"), 5));
-    roundtrip(dict! { "Enum": dict! { "Tuple" : varray!["four", 5] } });
-}
-
-#[itest]
-fn enum_named() {
-    roundtrip(Enum::Named {
-        data: String::from("data"),
-    });
-    roundtrip(dict! {
-        "Enum": dict!{ "Named": dict!{ "data": "data" } }
-    });
-}
-
-// ----------------------------------------------------------------------------------------------------------------------------------------------
-// Skipping of enums
-
-macro_rules! roundtrip_with_skip {
-    ($name_to:ident, $name_from:ident, $value:expr, $to_var:expr, $from_var:expr) => {
-        #[itest]
-        fn $name_to() {
-            let s = $value;
-            assert_eq!(s.to_variant(), $to_var.to_variant(),)
+fn enum_inty() {
+    roundtrip(EnumInty::A);
+    assert_eq!(EnumInty::A.to_godot(), 10);
+    roundtrip(EnumInty::B);
+    assert_eq!(EnumInty::B.to_godot(), 11);
+    roundtrip(EnumInty::C);
+    assert_eq!(EnumInty::C.to_godot(), 12);
+    roundtrip(EnumInty::D);
+    assert_eq!(EnumInty::D.to_godot(), 1);
+    roundtrip(EnumInty::E);
+    assert_eq!(EnumInty::E.to_godot(), 2);
```

Comment:
Instead of alternating, maybe group all the roundtrips and all the asserts, with a newline in between?
So we group by testing pattern and not by data.